internal/icinga2/client.go

```diff
@@ -132,17 +138,53 @@ func (client *Client) buildCommonEvent(ctx context.Context, host, service string
 		Name:      eventName,
 		URL:       eventUrl.String(),
 		Tags:      eventTags,
-		ExtraTags: eventExtraTags,
+		ExtraTags: extraTags,
 	}, nil
 }
 
+// refreshExtraTagsCache refreshes the client event extra tags cache store based on the given response result.
+// If the specified response result is of typeObjectCreated, it will return the just extracted event extra tags
+// (if available), otherwise always nil.
+func (client *Client) refreshExtraTagsCache(ctx context.Context, result *CheckableCreatedDeleted) (map[string]string, error) {
+	switch result.EventType {
+	case typeObjectDeleted:
+		// The checkable has just been deleted, so delete all existing extra tags from our cache store as well.
+		client.eventExtraTagsCache.Remove(result.ObjectName)
+	case typeObjectCreated:
+		extraTags := make(map[string]string)
+		queryResult, err := client.fetchCheckableCurrentState(ctx, result.GetHostName(), "")
+		if err != nil {
+			return nil, err
+		}
+		for _, hostGroup := range queryResult.Attrs.Groups {
+			extraTags["hostgroup/"+hostGroup] = ""
+		}
+
+		if result.ObjectType == "Service" {
+			queryResult, err := client.fetchCheckableCurrentState(ctx, result.GetHostName(), result.GetServiceName())
+			if err != nil {
+				return nil, err
+			}
+			for _, serviceGroup := range queryResult.Attrs.Groups {
+				extraTags["servicegroup/"+serviceGroup] = ""
+			}
+		}
+
+		client.eventExtraTagsCache.Add(result.ObjectName, extraTags)
```

Comment:
Shouldn't a `return extraTags, nil` be present at this point?

Reply:
🐘